test/integration/userFlow.test.js

```diff
@@ -0,0 +1,187 @@
+/* global beforeAll afterAll expect */
+import mongoose from 'mongoose';
+import { graphql } from 'graphql';
+import { pick } from 'lodash';
+import UserModel from '../../src/dataLayer/model/user';
+import { graphqlSchema } from '../../src/handler';
+
+beforeAll(async function beforeAllTests() {
+  await mongoose.connect(global.__MONGO_URI__);
+});
+
+afterAll(async function afterAllTests() {
+  await mongoose.disconnect();
+});
+
+const contextNoToken = global.mockedContextWithOutToken;
+const invalidContext = global.mockedContextWithInValidToken;
+const validContextCharlie = global.mockedContextWithValidTokenForCharlie;
+// language=GraphQL
+
+const createUserQuery = `
+mutation createUser {
+  createUser {
+    email
+    name
+  }
+}
+`;
+
+const expectedUserQuery = `
+    query {
+        getUser(email: "charlie@thebear.me") {
+            name
+            email
+        }
+    }
+  `;
+
+const expectedNoUserQuery = `
+    query {
+      getUser(email: "nowhere@tobe.seen") {
+        name
+        email
+      }
+    }
+  `;
+
+const malformedQuery = `
+    query {
+      getUser(email: "Ooops!") {
+        name
+        email
+      }
+    }
+  `;
+
+const rootValue = {};
+
+describe('createUser', () => {
+  it('should return null and an auth error without a token', done => {
+    expect.assertions(2);
+
+    graphql(graphqlSchema, createUserQuery, rootValue, contextNoToken)
+      .then(({ data, errors }) => {
+        expect(data.createUser).toMatchSnapshot('no user');
+        expect(errors).toMatchSnapshot('not logged in');
+        return;
+      })
+      .then(done)
+      .catch(done);
+  });
+
+  it('should return null and an auth error with an invalid token', done => {
+    expect.assertions(2);
+
+    graphql(graphqlSchema, createUserQuery, rootValue, invalidContext)
+      .then(({ data, errors }) => {
+        expect(data.createUser).toMatchSnapshot('no user');
+        expect(errors).toMatchSnapshot('no auth error');
+        return;
+      })
+      .then(done)
+      .catch(done);
+  });
+
+  it('should create a user by query', done => {
+    expect.assertions(2);
+
+    graphql(graphqlSchema, createUserQuery, rootValue, validContextCharlie)
+      .then(({ data, errors }) => {
+        expect(data.createUser).toMatchSnapshot('new user');
+        expect(errors).toMatchSnapshot('no errors');
+        return;
+      })
+      .then(done)
+      .catch(done);
+  });
+
+  it('should not create a second user for the same context', done => {
```

Comment:
I think we are testing that we are not allowed to create duplicate users, right? Should we use that in the description instead?